### ISSUE TITLE
Inbox sorting broke with invalid date

### ISSUE THREAD

**Issue by davidwagner** [mikeaboody/phishing-research#28]
Looks like Inbox's compare_emails() needs to be tweaked to be robust to bogus/invalid dates.

I suggest writing a little function to extract the date out of an email, or return None if you can't, and make sure it catches exceptions and returns None.  E.g., something like:

```
def get_time(email):
    try:
        t = email.utils.parsedate(email['Date'])
        if not t:
            return None
        return time.mktime(t)
    except ValueError:
        return None
    except OverflowError:
        return None
```

and then compare_emails() becomes something like:

```
def compare_emails(email1, email2):
    t1 = get_time(email1)
    t2 = get_time(email2)
    if t1 == t2:
        return 0
    if not t1 or t1 > t2:
        return 1
    return -1
```

(not tested).

Here's the traceback from Vern:

```
Traceback (most recent call last):
  File "phish_detector.py", line 218, in <module>
    detector.execute()
  File "phish_detector.py", line 197, in execute
    self.generate_features()
  File "phish_detector.py", line 169, in generate_features
    self.prep_features()
  File "phish_detector.py", line 160, in prep_features
    self.detectors
  File "/da/home/vern/phishing-research/common/generate_features.py", line 63, in __init__
    self.emails = inbox.Inbox(filename)
  File "/da/home/vern/phishing-research/common/inbox.py", line 14, in __init__  
    self.sort_emails()
  File "/da/home/vern/phishing-research/common/inbox.py", line 56, in sort_emails
    self.emails = sorted(self.emails, cmp=compare_emails)
  File "/da/home/vern/phishing-research/common/inbox.py", line 49, in compare_emails
    time1 = time.mktime(email.utils.parsedate(date_header1))
ValueError: year out of range
```

**Comment by davidwagner** [comment]
Oops, that compare_emails() code might not work quite right if t2 is None -- adjust logic accordingly and test.

**Comment by mikeaboody** [comment]
Consider t2 = None, and t1 = None. Then, since they're equal, it returns 0. Consider t2 = None, and t1 = something else. Then, it returns -1 just as expected since we want t1 to be before t2 if t2 is None. I remember talking about this in the meeting where we said we want all of the one's with no date or no parseable date to be put at the end, and this should do it right.

The actual error from this is a really weird error, as you mentioned, http://stackoverflow.com/questions/24487567/time-mktime-does-not-accept-year-before-1900. I'll fix it the exact way you suggested.

**Comment by mikeaboody** [comment]
Fixed and tested: 6f89a886329a711a6d1f9d50efcf4dbad87ddfb1
